data/blog/posts/2022/2022-07-04-how-to-automatically-add-return-type-declarations-without-breaking-your-code.md

```diff
@@ -163,30 +163,8 @@ And much more!
 
 Depending on what age your project is coming from, there is one requirement to make this work. You must use PHP 7.0, where [return type declarations](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.return-type-declarations) were added.
 
-## Try it Yourself
+In the project we tested these rules on, we had over 40 changed files in the first run. From zero to 40 files, quite impressive. And as you know, strict types [spread exponentially](/blog/2021/02/15/how-much-does-single-type-declaration-know).
 
-For each case above, there is one rector rule that handles this case. Add these 4 rules to your code and see for yourself:
+## Coming in Rector Enterprise
 
-```php
-use Rector\Config\RectorConfig;
-use Rector\CodeQuality\Rector\ClassMethod\ReturnTypeFromStrictScalarReturnExprRector;
-use Rector\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictBoolReturnExprRector;
-use Rector\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictNativeFuncCallRector;
-use Rector\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictNewArrayRector;
-
-return static function (RectorConfig $rectorConfig): void {
-    $rectorConfig->rules([
-        ReturnTypeFromStrictBoolReturnExprRector::class,
-        ReturnTypeFromStrictNativeFuncCallRector::class,
-        ReturnTypeFromStrictNewArrayRector::class,
-        ReturnTypeFromStrictScalarReturnExprRector::class,
```

Comment:
ReturnTypeFromStrictNativeFuncCallRector and ReturnTypeFromStrictScalarReturnExprRector seems not enterprise yet

currently, only these 2 that on enterprise which mentioned in blog post:

- ReturnTypeFromStrictBoolReturnExprRector
- ReturnTypeFromStrictNewArrayRector

Reply:
In the end, it will be better to keep those 2 to try it out and the rest make part of Rector Enterprise.